src/FileType_HDF5.f90

```diff
@@ -1153,6 +1162,109 @@ FUNCTION pathexists_HDF5FileType(thisHDF5File,path) RESULT(bool)
+      TYPE(StringType) :: path2
+
+      bool=.FALSE.
+
+      ! Make sure the object is initialized, and opened
+      REQUIRE(thisHDF5File%isinit)
+      REQUIRE(thisHDF5File%isOpen())
+
+      IF(PRESENT(path)) THEN
+        IF(pathexists_HDF5FileType(thisHDF5File,path)) THEN
+          path2=convertPath(path)
+          nelmts=1
+
+          !Get the data set ID
+          CALL h5dopen_f(thisHDF5File%file_id,TRIM(path2),dset_id,error)
+
+          !Get the data set creation property list
+          CALL h5dget_create_plist_f(dset_id,dcpl,error)
+
+          !Get the number of filters on the data set, and loop over
+          !them to find a compression filter
+          CALL h5pget_nfilters_f(dcpl,nfilters,error)
+          DO i=0,nfilters-1
+            CALL h5pget_filter_f(dcpl,i,flags,nelmts,cd_values, &
+              namelen,filter_name,filter_id,error)
+            bool=ANY(filter_id == (/H5Z_FILTER_DEFLATE_F, &
+              H5Z_FILTER_SZIP_F,H5Z_FILTER_NBIT_F,H5Z_FILTER_NBIT_F/))
+            IF(bool) EXIT
+          ENDDO
+        ENDIF
+      ELSE
+        bool=thisHDF5File%hasCompression
+      ENDIF
+#else
+      bool=.FALSE.
+#endif
+    ENDFUNCTION isCompressed_HDF5FileType
+!
+!-------------------------------------------------------------------------------
+!> @brief   Returns the chunk size of a dataset
+!> @param   thisHDF5File the HDF5FileType object to interrogate
+!> @param   path the path to a dataset in the file
+!> @param   cdims allocatabel array containing chunk sizes for each dimension
+!>          of dataset.
+!>
+!> If a dataset does not have chunking or does not exist, then cdims is
+!> returned unallocated.
+!>
+    SUBROUTINE getChunkSize_HDF5FileType(thisHDF5File,path,cdims)
+      CLASS(HDF5FileType),INTENT(INOUT) :: thisHDF5File
+      CHARACTER(LEN=*),INTENT(IN) :: path
+      INTEGER(HSIZE_T),ALLOCATABLE,INTENT(INOUT) :: cdims(:)
+
+#ifdef FUTILITY_HAVE_HDF5
+      INTEGER(SIK) :: error,ndims,layout
+      INTEGER(HID_T) :: dset_id,dspace_id,dcpl
+      TYPE(StringType) :: path2
+
+      IF(ALLOCATED(cdims)) DEALLOCATE(cdims)
+
+      ! Make sure the object is initialized, and opened
+      REQUIRE(thisHDF5File%isinit)
+      REQUIRE(thisHDF5File%isOpen())
+      IF(pathexists_HDF5FileType(thisHDF5File,path)) THEN
+        path2=convertPath(path)
+
+        !Get the data set ID, associated data space, and rank
+        CALL h5dopen_f(thisHDF5File%file_id,TRIM(path2),dset_id,error)
+        CALL h5dget_space_f(dset_id,dspace_id,error)
+        CALL h5sget_simple_extent_ndims_f(dspace_id,ndims,error)
+
+        !Get the data set creation property list
+        CALL h5dget_create_plist_f(dset_id,dcpl,error)
+
+        !Get the data space layout and chunk size
+        CALL h5pget_layout_f(dcpl,layout,error)
+        IF(layout == H5D_CHUNKED_F) THEN
+          ALLOCATE(cdims(ndims)); cdims=-1
+          CALL h5pget_chunk_f(dcpl,SIZE(cdims),cdims,error)
+        ENDIF
+      ENDIF
```

Comment:
Yeah... I'm torn about how to handle this...

It comes down to, do we expect the client code to want to call this on non-existent data sets?

If we don't want this to happen, then there should be a DBC ```REQUIRE``` here so the developer knows not to do that.

However, I can see potential cases where we might end up executing in a mode where somehow the code constructs a garbled name that is expected to exist but for some other bug elsewhere it doesn't, and this triggers it...

Alright, over thinking it. Simplifying:
* There is presently one client for this routine: ```postRead```
  * which is guaranteed to have read a dataset that exists or the error is caught by preRead
 * Therefore the contract for ```isCompressed``` is that it should only be called for datasets that exist
    * Meaning I should add a ```REQUIRE```

Do you agree with that line of thinking Aaron?

Reply:
My only change to that line of thinking would be that an actual error might be more useful in debugging than a REQUIRE since we could print the non-existent dataset.  But I don't care too much either way.